pkg/kubectl/rolling_updater.go

```diff
@@ -395,6 +397,12 @@ func (r *RollingUpdater) pollForReadyPods(interval, timeout time.Duration, oldRc
 // Existing controllers are validated to ensure their sourceIdAnnotation
 // matches sourceId; if there's a mismatch, an error is returned.
 func (r *RollingUpdater) getOrCreateTargetControllerWithClient(controller *api.ReplicationController, sourceId string) (*api.ReplicationController, bool, error) {
+	// Only create but not get when the controller has not name but generatedName,
+	// Otherwise it'll complain "no resource name".
+	if len(controller.Name) == 0 && len(controller.GenerateName) > 0 {
+		newRc, err := r.createTargetController(controller, sourceId)
```

Comment:
This bypasses the check for Replicas <= 0 below - is that correct?

Could you instead do something like:

```
existing, err := testExists(controller)
if err != nil {
    //...
}
if controller.Spec.Replicas <= 0 {
    //...
}
newRc, err := r.createTargetController(controller, sourceId)
```

and

```
func testExists(controller) {
    if len(controller.Name) == 0 && len(controller.GenerateName) > 0 {
        return false, nil
    }
    return r.c.ReplicationControllers(controller.Namespace).Get(controller.Name)
}
```